### ISSUE TITLE
manually created cluster and state syncing oddities.

### ISSUE THREAD

**Issue by hsyed** [kubernetes/kops#3457]
### Issue

We have a 3xHA fully private cluster with no bastion created in a shared VPC. The utility and private subnets were created manually. The VPC is meant to house multiple clusters, it has one other cluster in it at the moment. The cluster bootstrapped fine and it's working fine. 

I added a new SG to the worker nodes and `output 1` (see below) is what I got when I planned the update. The second cluster in the VPC is going to be recreated so I went ahead and applied it.

I expected kops to have now synced the state of the cluster but my colleague got `output 2` when he ran a `kops update`.

I expect:

* kops not to want to rename and retag these instances as it should know it was created in existing infrastructure.
* kops not to show entries for which it is not going to modify infrastructure. 

### output 1:

```
 DNSName/api.devb.k8s.foo.bar.com
        Lifecycle                <nil> -> Sync

  InternetGateway/devb.k8s.foo.bar.com
        Name                     dev_.k8s.foo.bar.com -> devb.k8s.foo.bar.com

  LaunchConfiguration/nodes.devb.k8s.foo.bar.com
        SecurityGroups           [id:sg-abcd] -> [name:nodes.devb.k8s.foo.bar.com id:sg-abcd, name:sg-abcd id:sg-abcd]

  LoadBalancer/api.devb.k8s.foo.bar.com
        Lifecycle                <nil> -> Sync

  LoadBalancerAttachment/api-master-eu-west-1a
        Lifecycle                <nil> -> Sync

  LoadBalancerAttachment/api-master-eu-west-1b
        Lifecycle                <nil> -> Sync

  LoadBalancerAttachment/api-master-eu-west-1c
        Lifecycle                <nil> -> Sync

  Subnet/eu-west-1a.devb.k8s.foo.bar.com
        Name                     devb.pvt.eu-west-1a.k8s.foo.bar.com -> eu-west-1a.devb.k8s.foo.bar.com
        Tags                     {Name: devb.pvt.eu-west-1a.k8s.foo.bar.com, KubernetesCluster: devb.k8s.foo.bar.com} -> {KubernetesCluster: devb.k8s.foo.bar.com, Name: eu-west-1a.devb.k8s.foo.bar.com, kubernetes.io/cluster/devb.k8s.foo.bar.com: shared}

  Subnet/eu-west-1b.devb.k8s.foo.bar.com
        Name                     devb.pvt.eu-west-1b.k8s.foo.bar.com -> eu-west-1b.devb.k8s.foo.bar.com
        Tags                     {Name: devb.pvt.eu-west-1b.k8s.foo.bar.com, KubernetesCluster: devb.k8s.foo.bar.com} -> {KubernetesCluster: devb.k8s.foo.bar.com, Name: eu-west-1b.devb.k8s.foo.bar.com, kubernetes.io/cluster/devb.k8s.foo.bar.com: shared}

  Subnet/eu-west-1c.devb.k8s.foo.bar.com
        Name                     devb.pvt.eu-west-1c.k8s.foo.bar.com -> eu-west-1c.devb.k8s.foo.bar.com
        Tags                     {Name: devb.pvt.eu-west-1c.k8s.foo.bar.com, KubernetesCluster: devb.k8s.foo.bar.com} -> {KubernetesCluster: devb.k8s.foo.bar.com, Name: eu-west-1c.devb.k8s.foo.bar.com, kubernetes.io/cluster/devb.k8s.foo.bar.com: shared}

  Subnet/utility-eu-west-1a.devb.k8s.foo.bar.com
        Name                     devb.pub.eu-west-1a.k8s.foo.bar.com -> utility-eu-west-1a.devb.k8s.foo.bar.com
        Tags                     {Name: devb.pub.eu-west-1a.k8s.foo.bar.com, KubernetesCluster: devb.k8s.foo.bar.com} -> {KubernetesCluster: devb.k8s.foo.bar.com, Name: utility-eu-west-1a.devb.k8s.foo.bar.com, kubernetes.io/cluster/devb.k8s.foo.bar.com: shared}

  Subnet/utility-eu-west-1b.devb.k8s.foo.bar.com
        Name                     devb.pub.eu-west-1b.k8s.foo.bar.com -> utility-eu-west-1b.devb.k8s.foo.bar.com
        Tags                     {KubernetesCluster: devb.k8s.foo.bar.com, Name: devb.pub.eu-west-1b.k8s.foo.bar.com} -> {KubernetesCluster: devb.k8s.foo.bar.com, Name: utility-eu-west-1b.devb.k8s.foo.bar.com, kubernetes.io/cluster/devb.k8s.foo.bar.com: shared}

  Subnet/utility-eu-west-1c.devb.k8s.foo.bar.com
        Name                     devb.pub.eu-west-1c.k8s.foo.bar.com -> utility-eu-west-1c.devb.k8s.foo.bar.com
        Tags                     {KubernetesCluster: devb.k8s.foo.bar.com, Name: devb.pub.eu-west-1c.k8s.foo.bar.com} -> {kubernetes.io/cluster/devb.k8s.foo.bar.com: shared, KubernetesCluster: devb.k8s.foo.bar.com, Name: utility-eu-west-1c.devb.k8s.foo.bar.com}

  VPC/devb.k8s.foo.bar.com
        Name                     dev_.k8s.foo.bar.com -> devb.k8s.foo.bar.com
        Tags                     {Name: dev_.k8s.foo.bar.com} -> {KubernetesCluster: devb.k8s.foo.bar.com, Name: devb.k8s.foo.bar.com, kubernetes.io/cluster/devb.k8s.foo.bar.com: shared}
```

### output 2

```
Will modify resources:
  DNSName/api.devb.k8s.foo.bar.com
      Lifecycle                <nil> -> Sync

  InternetGateway/devb.k8s.foo.bar.com
      Name                     dev_.k8s.foo.bar.com -> devb.k8s.foo.bar.com

  LoadBalancer/api.devb.k8s.foo.bar.com
      Lifecycle                <nil> -> Sync

  LoadBalancerAttachment/api-master-eu-west-1a
      Lifecycle                <nil> -> Sync

  LoadBalancerAttachment/api-master-eu-west-1b
      Lifecycle                <nil> -> Sync

  LoadBalancerAttachment/api-master-eu-west-1c
      Lifecycle                <nil> -> Sync

  Subnet/eu-west-1a.devb.k8s.foo.bar.com
      Name                     devb.pvt.eu-west-1a.k8s.foo.bar.com -> eu-west-1a.devb.k8s.foo.bar.com
      Tags                     {Name: devb.pvt.eu-west-1a.k8s.foo.bar.com, KubernetesCluster: devb.k8s.foo.bar.com} -> {kubernetes.io/cluster/devb.k8s.foo.bar.com: shared, KubernetesCluster: devb.k8s.foo.bar.com, Name: eu-west-1a.devb.k8s.foo.bar.com}

  Subnet/eu-west-1b.devb.k8s.foo.bar.com
      Name                     devb.pvt.eu-west-1b.k8s.foo.bar.com -> eu-west-1b.devb.k8s.foo.bar.com
      Tags                     {Name: devb.pvt.eu-west-1b.k8s.foo.bar.com, KubernetesCluster: devb.k8s.foo.bar.com} -> {KubernetesCluster: devb.k8s.foo.bar.com, Name: eu-west-1b.devb.k8s.foo.bar.com, kubernetes.io/cluster/devb.k8s.foo.bar.com: shared}

  Subnet/eu-west-1c.devb.k8s.foo.bar.com
      Name                     devb.pvt.eu-west-1c.k8s.foo.bar.com -> eu-west-1c.devb.k8s.foo.bar.com
      Tags                     {Name: devb.pvt.eu-west-1c.k8s.foo.bar.com, KubernetesCluster: devb.k8s.foo.bar.com} -> {Name: eu-west-1c.devb.k8s.foo.bar.com, kubernetes.io/cluster/devb.k8s.foo.bar.com: shared, KubernetesCluster: devb.k8s.foo.bar.com}

  Subnet/utility-eu-west-1a.devb.k8s.foo.bar.com
      Name                     devb.pub.eu-west-1a.k8s.foo.bar.com -> utility-eu-west-1a.devb.k8s.foo.bar.com
      Tags                     {Name: devb.pub.eu-west-1a.k8s.foo.bar.com, KubernetesCluster: devb.k8s.foo.bar.com} -> {KubernetesCluster: devb.k8s.foo.bar.com, Name: utility-eu-west-1a.devb.k8s.foo.bar.com, kubernetes.io/cluster/devb.k8s.foo.bar.com: shared}

  Subnet/utility-eu-west-1b.devb.k8s.foo.bar.com
      Name                     devb.pub.eu-west-1b.k8s.foo.bar.com -> utility-eu-west-1b.devb.k8s.foo.bar.com
      Tags                     {KubernetesCluster: devb.k8s.foo.bar.com, Name: devb.pub.eu-west-1b.k8s.foo.bar.com} -> {KubernetesCluster: devb.k8s.foo.bar.com, Name: utility-eu-west-1b.devb.k8s.foo.bar.com, kubernetes.io/cluster/devb.k8s.foo.bar.com: shared}

  Subnet/utility-eu-west-1c.devb.k8s.foo.bar.com
      Name                     devb.pub.eu-west-1c.k8s.foo.bar.com -> utility-eu-west-1c.devb.k8s.foo.bar.com
      Tags                     {Name: devb.pub.eu-west-1c.k8s.foo.bar.com, KubernetesCluster: devb.k8s.foo.bar.com} -> {KubernetesCluster: devb.k8s.foo.bar.com, Name: utility-eu-west-1c.devb.k8s.foo.bar.com, kubernetes.io/cluster/devb.k8s.foo.bar.com: shared}

  VPC/devb.k8s.foo.bar.com
      Name                     dev_.k8s.foo.bar.com -> devb.k8s.foo.bar.com
      Tags                     {Name: dev_.k8s.foo.bar.com} -> {KubernetesCluster: devb.k8s.foo.bar.com, Name: devb.k8s.foo.bar.com, kubernetes.io/cluster/devb.k8s.foo.bar.com: shared}

Must specify --yes to apply changes
```

**Comment by faheem-cliqz** [comment]
This might be related to  #3450. I am also seeing the same behavior with kops 1.7.0 and k8s 1.7.6

**Comment by chrislovecnm** [comment]
1. You need to have the shared tag and corrxf k8s tags on the subnets
2. You can skip the network phase by only running iam and cluster phases

kops update cluster --phase iam
Then
kops update cluster --phase cluster

**Comment by hsyed** [comment]
@chrislovecnm 

I can update the tags in private subnets since they belong to the cluster in question. This gets rid of the modify entries for those. 


For the shared resources (utility subnets, VPC and IGW) I updated them so that they do not contain a `KubernetesCluster` cluster tag and I added `KubernetesCluster: devb.k8s.foo.bar.com` to it. I now expect the modify entry to disappear however It still wants to update the Name tag and add a `KubernetesCluster` tag.
